vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,197 @@
+            @RequestParam(name = "format", required = false) String format,
+            @RequestParam(name = "referenceName", required = false) String referenceName,
+            @RequestParam(name = "species", required = false) String species,
+            @RequestParam(name = "start", required = false) Integer start,
+            @RequestParam(name = "end", required = false) Integer end,
+            @RequestParam(name = "fields", required = false) List<String> fields,
+            @RequestParam(name = "tags", required = false, defaultValue = "") String tags,
+            @RequestParam(name = "notags", required = false, defaultValue = "") String notags,
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        if (!VCF.equals(format)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Collections.singletonMap("htsget",
+                            new HtsGetError("UnsupportedFormat", "Specified format is not supported by this server")));
+        }
+
+        String dbName = "eva_" + species;
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections.singletonList(referenceName));
+
+        int blockSize = Integer.parseInt(evaProperties.getProperty("eva.htsget.blocksize"));
+
+        VariantExporterController controller = new VariantExporterController(dbName, Arrays.asList(id.split(",")),
+                                                                             evaProperties,
+                                                                             queryParameters, blockSize);
+        if (!controller.validateSpecies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Collections.singletonMap("htsget",
+                            new HtsGetError("InvalidInput", "The requested species is not available")));
+        }
+        if (!controller.validateStudies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Collections.singletonMap("htsget",
+                            new HtsGetError("InvalidInput", "The requested study(ies) is not available")));
+        }
+        if (start != null && referenceName == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(Collections.singletonMap("htsget",
+                    new HtsGetError("InvalidInput", "Reference name is not specified when start is specified")));
+        }
+
+        if (start == null) {
+            start = controller.getCoordinateOfFirstVariant(referenceName);
+        }
+        if (end == null) {
+            end = controller.getCoordinateOfLastVariant(referenceName);
+        }
+
+        if (end <= start) {
```

Comment:
@cyenyxe, what do you think about this possible scenary:

Imagine an user has done a valid petition: chromosome 1, start: 1.000.000, end: empty, study: X.
The study X has variants just in the regions 200.000 to 800.000. The controller returns 800.000 when we call to getCoordinateOfLastVariant().  Like 800.000 < 1.000.000 , an error message is sent "The requested range cannot be satisfied"

This is the same error returned if the user makes an incorrect request like chromosome 1, start: 1.000.000, end: 800.000. Wouldn't it be better if we return a different message? We could return also an empty list of urls for retrieving the VCF body parts, but that would be even more confusing ....

Reply:
done in 22b1a3f926035ac77c1716ece13cf6b53829a1a7

Reply:
I feel a comment would clarify what this condition is exactly for, because there is a very similar one above. Something inspired on @pabarcgar's text: 

```
Applies to valid requests such as chromosome 1, start: 1.000.000, end: empty.
If variants exist only in region 200.000 to 800.000, the call to getCoordinateOfLastVariant() will return 800.000.
Given that 800.000 < 1.000.000, no region can be found.
```